src/common/utility_mac.mm

```diff
@@ -61,6 +61,25 @@
 static Result<void, QString> writePlistToFile(NSString *plistFile, NSDictionary *plist)
 {
     NSError *error = nil;
+
+    // Check if the directory exists. On a fresh installation for example, the directory does not
+    // exist, so writing the plist file below will fail.
+    QDir plistDir = QFileInfo(QString::fromNSString(plistFile)).dir();
```

Comment:
Why the extra step through QFileInfo?

Reply:
To get the directory from the file. The input is something like `<homedir>/Library/LaunchAgents/foo.plist`, and I want to check if the `LaunchAgents` dir exists. Is there a better way to do that?

Reply:
Ah yes